plugins/apps/threatbus_misp/threatbus_misp/message_mapping.py

```diff
@@ -106,69 +204,122 @@ def is_whitelisted(misp_msg: dict, filter_config: List[Dict]):
     return False
 
 
-def map_to_internal(misp_attribute: dict, action: str, logger=None):
+def stix2_indicator_id(attr_uuid: str) -> str:
+    """
+    Converts any given MISP attribute UUID to a STIX-2 indicator ID.
+    @param attr_id The MISP attribute ID to convert
+    @return a valid STIX-2 Indicator ID
+    """
+    return f"indicator--{attr_uuid}"
+
+
+def misp_id(stix2_indicator_uuid: str) -> str:
     """
-    Maps the given MISP attribute to the threatbus Intel format. Discards all
-    messages that do not match the given filter, if any.
-    @param misp_attribute The MISP attribute to map
+    Converts any given STIX-2 indicator ID to a valid UUID.
+    @param stix2_indicator_uuid The STIX-2 Indicator ID to convert
+    @return a valid uuid
+    """
+    return stix2_indicator_uuid[len("indicator--") :]
+
+
+def attribute_to_stix2_indicator(
+    misp_attribute: dict, action: str, logger
+) -> Union[None, Update, Indicator]:
+    """
+    Maps the given MISP attribute to a STIX-2 cyber observable object.
+    @param misp_attribute The MISP attribute to map to STIX-2
     @param action A string from MISP, describing the action for the attribute
-        (either 'add' or 'delete')
-    @return The mapped intel item or None
+        (either 'add', 'edit', or 'delete')
```

Comment:
This function is in spirit a parser. I recommend verifying that the parameters contain allowed values early on and switching into the error path if they don't. I guess that means throwing an exception in python land. This would free you from looking at the arguments again after the function returned null at the call site.